src/bp/ui-studio/src/web/views/OneFlow/sidePanel/Library/index.tsx

```diff
@@ -1,186 +1,220 @@
+    const original = props.entities.find(x => x.id === entityId)
+    const name = getNextName(entityId, props.entities)
+
+    await createEntity({ ...original, id: name, name })
   }
 
-  const updateEntity = async (targetEntityId: string, entity: NLU.EntityDefinition) => {
-    await axios.post(`${window.BOT_API_PATH}/nlu/entities/${targetEntityId}`, entity)
-    const idx = entities.findIndex(ent => ent.name === entity.name)
-    setEntities([...entities.slice(0, idx), entity, ...entities.slice(idx + 1)])
+  const createEntity = async entity => {
+    await axios.post(`${window.BOT_API_PATH}/nlu/entities`, entity)
+    props.refreshEntities()
+    props.setActiveFormItem({ type: 'entity', data: entity })
   }
 
-  const renderTable = (title: string, items: TableItem[]) => {
-    return (
-      <table
-        className={cx(
-          style.table,
-          'bp3-html-table bp3-html-table-striped bp3-html-table-bordered .bp3-html-table-condensed'
-        )}
-      >
-        <thead>
-          <tr>
-            <th>{title}</th>
-            <th></th>
-            <th></th>
-          </tr>
-        </thead>
-        <tbody>{renderTableRows(items)}</tbody>
-      </table>
-    )
+  const deleteEntity = async (entityId: string) => {
+    if (await confirmDialog(lang.tr('studio.library.confirmDeleteEntity'), { acceptLabel: lang.tr('delete') })) {
+      props.deleteEntity(entityId)
+      props.refreshEntities()
+    }
   }
-  const renderTableRows = (items: TableItem[]) => {
+
+  const handleContextMenu = (element: NodeData) => {
+    const { id, type } = element as NodeData
+
+    if (id === type) {
+      return
+    }
+
     return (
       <Fragment>
-        {items &&
-          items?.map((item, i) => (
-            <tr key={i}>
-              <td>
-                <Button text={item.label} onClick={item.click} />
-              </td>
-              <td>
-                <Button icon="edit" onClick={item.edit} />
-              </td>
-              <td>
-                <Button icon="delete" onClick={item.delete} />
-              </td>
-            </tr>
-          ))}
+        <MenuItem id="btn-duplicate" label={lang.tr('duplicate')} onClick={() => duplicateEntity(id)} />
+        <MenuItem id="btn-delete" label={lang.tr('delete')} intent={Intent.DANGER} onClick={() => deleteEntity(id)} />
       </Fragment>
     )
   }
 
-  const renderVariableTypes = () => {
-    return (
-      <section>
-        {renderTable(
-          'Variable Types',
-          entities
-            .filter(x => x.type !== 'system')
-            .map<TableItem>(x => ({
-              label: x.name,
-              click: () => {
-                setCurrentEntity(x)
-              },
-              edit: () => {
-                setCurrentNamingEntity(x)
-              },
-              delete: async () => {
-                await deleteEntity(x)
-              }
-            }))
+  const printTree = (item: NodeData, level, parentId = '') => {
```

Comment:
do we really need a tree view  since we have a single level deep ?

Reply:
I want to keep the form the same way we do for topics. Same logic, same behavior (eg: to rename saved workflows), and if we ever need a second level, everything's already there